packages/components/src/SegmentedControl/SegmentedControl.tsx

```diff
@@ -0,0 +1,47 @@
+import React, {
+  Children,
+  type PropsWithChildren,
+  useEffect,
+  useRef,
+} from "react";
+import styles from "./SegmentedControl.css";
+import { SegmentedControlProvider } from "./SegmentedControlProvider";
+import { SegmentedControlOption } from "./SegmentedControlOption";
+
+interface SegmentedControlProps<T> extends PropsWithChildren {
+  readonly selectedOption: T;
+  readonly onSelectOption: (view: T) => void;
+  readonly defaultOption: T;
+}
+
+export function SegmentedControl<T>({
+  onSelectOption,
+  defaultOption,
+  children,
+}: SegmentedControlProps<T>) {
+  const container = useRef<HTMLDivElement>(null);
+  useEffect(() => {
+    const length = Children.count(children);
+
+    if (container.current) {
+      container.current.style.setProperty(
+        "--segmentedControl--option-count",
+        String(length),
+      );
+    }
+  }, [children]);
```

Comment:
Children are technically always considered unstable so you may just want to remove either the dep array or just count the length and make it the dep array value.

Reply:
This way will also avoid the issue if someone dynamically changes the contents. Rare, but shouldn't assume its not going to occur.

Reply:
Forgot I had a better solution below. https://github.com/GetJobber/atlantis/pull/1966#discussion_r1690678615